src/uwtools/config/formats/base.py

```diff
@@ -46,7 +46,9 @@ def __repr__(self) -> str:
         """
         Returns the string representation of a Config object.
         """
-        return json.dumps(self.data)
+        s = StringIO()
+        yaml.dump(self.data, s)
+        return s.getvalue()
```

Comment:
`json.dumps()` was handy because of PyYAML's (as far as I know) lack of a dump-to-string function, but `json` has no idea how to handle a `UWYAMLRemove` object and raises an exception when it encounters it. `yaml`, however, has a representer registered for that type, so using `yaml.dump()` with a `StringIO` object does the trick.